pkg/apis/pipeline/pod/template.go

```diff
@@ -251,6 +307,24 @@ func MergeAAPodTemplateWithDefault(tpl, defaultTpl *AAPodTemplate) *AAPodTemplat
 	}
 }
 
+// mergeMaps takes two maps with the same key and value types
+// and merges them, giving priority to the items in the override map.
+func mergeMaps[K comparable, V any](base, overrides map[K]V) map[K]V {
+	result := make(map[K]V)
```

Comment:
may be we can just add the overwrite to the base and done with it, instead of creating new and return that

Reply:
I don't think overwriting the base is safe. The base/default can be reused. We should only change tpl.